### ISSUE TITLE
Webhooks fail when multiple inserts are done

### ISSUE THREAD

**Issue by devdras** [supabase/supabase#11560]
# Bug report

## Describe the bug

I have a webhook configured to watch a table for inserts. On an insert a POST request should be fired to an API hosted on Vercel. 
This works when one insert operation is carried out. If multiple inserts are carried out: .insert([{message: 1}, {message: 2}, {message: 3}]) then the webhooks aren't fired at all. If instead we carry out multiple inserts by calling the Supabase API multiple times then the webhook is fired for only some of the inserts. 

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Configure a webhook on a table to watch for insert operations and call an external API
2. Use the Supabase JS library to perform multiple inserts like: `let { data, error } = await supabase
              .from("example_table")
              .insert([{message: 1}, {message: 2}, {message: 3}, {message: 4}, {message: 5}, {message: 6}, etc ])
              .select("message");`

4. Check logs for external API to confirm it didn't receive any requests
5. See error

Something to note is that I was inserting 32 things at once.

## Expected behavior

For each insert, I would expect the external API to be called. It would also help if there was a log in Supabase that showed what happens when the webhook is run.

## System information

- OS: N/A
- Browser (if applies): N/A
- Version of supabase-js: 2.0.0-rc.8
- Version of Node.js: v16.17.0 on desktop, 18.x on Vercel (per their docs)

**Comment by danrasmuson** [comment]
Thank you for posting this. I'm receiving a similar error.

Surprisingly when I create 10 records at the same time it fires the function for each 10 records...

```
INSERT INTO public.profiles (id, first_name, last_name, stripe_customer_id, email)
SELECT u.id, NULL, NULL, NULL, u.email
FROM auth.users u
LEFT JOIN public.profiles p ON u.id = p.id
WHERE p.id IS NULL
LIMIT 10;
```

However, if I fire the function for 100 records...

```
INSERT INTO public.profiles (id, first_name, last_name, stripe_customer_id, email)
SELECT u.id, NULL, NULL, NULL, u.email
FROM auth.users u
LEFT JOIN public.profiles p ON u.id = p.id
WHERE p.id IS NULL
LIMIT 100;
```

It fails

**Comment by devdras** [comment]
Yup, basically the webhook behaviour is flakey. I haven't tried it recently. I wanted to use this to build an event driven architecture but can't without it being reliable. Ended up using QStash as a replacement for the webhook. Would love Supabase to have some kind of full featured event queue like SQS/Eventbridge/Service Bus.
